### PR TITLE
⚠️ Update management cluster minimum Kubernetes version to v1.19.1

### PR DESCRIPTION
**Pull request by vincepri**
In anticipation of controller-runtime v0.7 and v1alpha4, these changes
raise the minimum Kubernetes version required for the management cluster
to v1.19.1, it also updates the e2e tests accordingly.

/milestone v0.4.0

### CODE REVIEW

File: cmd/clusterctl/cmd/config_cluster.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ var configClusterClusterCmd = &cobra.Command{
 		clusterctl config cluster my-cluster --target-namespace=foo
 
 		# Generates a configuration file for creating workload clusters with a specific Kubernetes version.
-		clusterctl config cluster my-cluster --kubernetes-version=v1.16.0
+		clusterctl config cluster my-cluster --kubernetes-version=v1.19.1
```
[the remark this answers is not in the document]
Comment (vincepri) [reply cmd/clusterctl/cmd/config_cluster.go:74]: That's ok, was hoping to suggest and provide good examples (running one of the latest versions)

File: test/e2e/config/docker.yaml [hunk 2]
```diff
@@ -73,11 +73,11 @@ providers:
   - sourcePath: "../data/infrastructure-docker/cluster-template-machine-pool.yaml"
 
 variables:
-  KUBERNETES_VERSION: "v1.18.2"
+  KUBERNETES_VERSION: "v1.19.1"
```
[the remark this answers is not in the document]
Comment (vincepri) [reply test/e2e/config/docker.yaml:76]: Done

**Comment by fabriziopandini** [comment]
/hold
```
failed to run clusterctl init
Unexpected error:
    <*errors.fundamental | 0xc000838260>: {
        msg: "unsupported management cluster server version: v1.17.2 - minimum required version is v1.19.1",
        stack: [0x16af69a, 0x169f666, 0x16bd777, 0x16c3f55, 0x16c6ab8, 0x174f26c, 0x174f951, 0x4d7a07, 0x4d6ef9, 0x7c6b38, 0x7c3983, 0x7c3597, 0x7c4c47, 0x7c48a5, 0x7cea9f, 0x7ce56b, 0x7d0c46, 0x7d3418, 0x7d314d, 0x174e2b7, 0x52864f, 0x471ee1],
    }
    unsupported management cluster server version: v1.17.2 - minimum required version is v1.19.1
occurred
```

**Comment by vincepri** [comment]
@fabriziopandini This should be good to go!

**Pull request merged**